include/internal/refcount.h

```diff
@@ -100,17 +100,29 @@ static __inline int CRYPTO_DOWN_REF(volatile int *val, int *ret, void *lock)
     return 1;
 }
 #   else
-#    pragma intrinsic(_InterlockedExchangeAdd)
+#    if !defined(_WIN32_WCE)
+#     pragma intrinsic(_InterlockedExchangeAdd)
+#     define INTERLOCKED_EXCHANGE_ADD _InterlockedExchangeAdd
+#    else
+#     if _WIN32_WCE >= 0x600
+       extern long __cdecl _InterlockedExchangeAdd(long volatile*, long);
+#      define INTERLOCKED_EXCHANGE_ADD _InterlockedExchangeAdd
+#     else
+       // under Windows CE we still have old-style Interlocked* functions
+       extern long __cdecl InterlockedExchangeAdd(long volatile*, long);
+#      define INTERLOCKED_EXCHANGE_ADD InterlockedExchangeAdd
```

Comment:
I would prefer if you would have only a single `#define` here (and remove the other two) and revert the changes in `CRYPTO_UP_REF()` and `CRYPTO_DOWN_REF()`.
```
#      define _InterlockedExchangeAdd InterlockedExchangeAdd
```